### PR TITLE
Add a setPerTryIdleTimeoutSeconds method to EngineBuilder.

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Add a setPerTryIdleTimeoutSeconds method to EngineBuilder.
Add tests for setPerTryIdleTimeoutSeconds and setStreamIdleTimeoutSeconds.
Minor cleanup of envoy_config_test.cc.

Part of #2498 

Risk Level: Low
Testing: Added unit tests
Docs Changes: N/A
Release Notes: Updated version_history.txt

### CODE REVIEW

File: library/cc/engine_builder.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ class EngineBuilder {
   EngineBuilder& setAppId(const std::string& app_id);
   EngineBuilder& setDeviceOs(const std::string& app_id);
   EngineBuilder& setStreamIdleTimeoutSeconds(int stream_idle_timeout_seconds);
+  EngineBuilder& setPerTryIdleTimeoutSeconds(int per_try_idle_timeout_seconds);
```
Comment (Augustyniak) [inline library/cc/engine_builder.h:41]: our Kotlin and Swift engines use `add...` naming but the c++ seems to prefer `set...` one... I guess it's fine to keep using `set` in here (which I personally prefer)

**Review by Augustyniak — COMMENTED** [review]
Looks good! Can we update version_history file?

**Comment by RyanTheOptimist** [comment]
> Looks good! Can we update version_history file?

Whoops! Done.

**Review by Augustyniak — APPROVED** [review]
>
>>Looks good! Can we update version_history file?

>Whoops! Done.

Hopefully @jpsim did not notice

File: test/cc/unit/envoy_config_test.cc [hunk 2]
```diff
@@ -78,9 +78,9 @@ TEST(TestConfig, ConfigIsValid) {
 
 #if !defined(__APPLE__)
 TEST(TestConfig, SetUseDnsCAresResolver) {
-  auto engine_builder = EngineBuilder();
+  EngineBuilder engine_builder;
```
Comment (Augustyniak) [inline test/cc/unit/envoy_config_test.cc:81]: how does this work?

**Pull request merged**